### PR TITLE
Make config write maintain comments

### LINKED ISSUE #605: config command overwrites config
What I expected to happen was to change only the runtime-endpoint in the config.

But what happened was it generated an entirely new config with defaults overwriting the prior config.
```
mike@mike-VirtualBox:~/go/src/k8s.io/kubernetes$ cat /etc/crictl.yaml
# important note: do not delete me
runtime-endpoint: unix:///run/containerd/containerd.sock

mike@mike-VirtualBox:~/go/src/k8s.io/kubernetes$ sudo crictl config runtime-endpoint unix:///run/crio/crio.sock

mike@mike-VirtualBox:~/go/src/k8s.io/kubernetes$ cat /etc/crictl.yaml
runtime-endpoint: unix:///run/crio/crio.sock
image-endpoint: ""
timeout: 0
debug: false
```

Suggest a --default for generating a new config, and modify current behavior to not delete comment lines blank lines other text in the config.

### PR DESCRIPTION
**Pull request by hickeyma**
This PR maintains comments in the `crictl` config file when writing to file. 

Closes #605

### CODE REVIEW

File: pkg/common/file.go [hunk 1]
```diff
@@ -20,17 +20,20 @@ import (
...
 type Config struct {
-	RuntimeEndpoint string `yaml:"runtime-endpoint"`
-	ImageEndpoint   string `yaml:"image-endpoint"`
-	Timeout         int    `yaml:"timeout"`
-	Debug           bool   `yaml:"debug"`
+	RuntimeEndpoint string     //`yaml:"runtime-endpoint"`
+	ImageEndpoint   string     //`yaml:"image-endpoint"`
+	Timeout         int        //`yaml:"timeout"`
+	Debug           bool       //`yaml:"debug"`
+	yamlData        *yaml.Node //YAML representation of config
```
Comment (saschagrunert) [inline pkg/common/file.go:36]: Can we remove those comments if not needed?
```suggestion
	RuntimeEndpoint string
	ImageEndpoint   string
	Timeout         int
	Debug           bool
	yamlData        *yaml.Node //YAML representation of config
```

File: pkg/common/file.go [hunk 2]
```diff
@@ -40,23 +43,122 @@ func ReadConfig(filepath string) (*Config, error) {
...
 	return ioutil.WriteFile(filepath, data, 0o644)
 }
+
+// Extracts config options from the yaml data which is loaded from file
+func getConfigOptions(yamlData yaml.Node) (*Config, error) {
+	config := Config{}
+	config.yamlData = &yamlData
+	contentLen := len(yamlData.Content[0].Content)
+
+	// YAML respreentation contains 2 yaml.ScalarNodes per config option
```
Comment (saschagrunert) [inline pkg/common/file.go:82]:
```suggestion
	// YAML representation contains 2 yaml.ScalarNodes per config option
```

File: pkg/common/file.go [hunk 3]
```diff
@@ -40,23 +43,122 @@ func ReadConfig(filepath string) (*Config, error) {
...
 		return err
 	}
 	return ioutil.WriteFile(filepath, data, 0o644)
 }
+
+// Extracts config options from the yaml data which is loaded from file
+func getConfigOptions(yamlData yaml.Node) (*Config, error) {
+	config := Config{}
+	config.yamlData = &yamlData
+	contentLen := len(yamlData.Content[0].Content)
```
Comment (saschagrunert) [inline pkg/common/file.go:80]: What if `yamlData.Content == nil` or  `yamlData.Content[0].Content == nil`?

File: pkg/common/file.go (hunk 1 above)
Comment (hickeyma) [reply pkg/common/file.go:36]: Sure, removed.

File: pkg/common/file.go (hunk 2 above)
Comment (hickeyma) [reply pkg/common/file.go:82]: fixed

File: pkg/common/file.go (hunk 3 above)
Comment (hickeyma) [reply pkg/common/file.go:85]: This is currently a WIP. It is now unfortunately writing the tags of each value to file. I have opened issue https://github.com/go-yaml/yaml/issues/619 in go-yaml project.  Maybe someone might know also whats up?

**Comment by hickeyma** [comment]
@saschagrunert Thanks for reviewing. I am still working on the empty YAML node comment. Setting PR to draft until resolve the issue explained in https://github.com/kubernetes-sigs/cri-tools/pull/615#discussion_r431968335.

**Comment by saschagrunert** [comment]
> @saschagrunert Thanks for reviewing. I am still working on the empty YAML node comment. Setting PR to draft until resolve the issue explained in [#615 (comment)](https://github.com/kubernetes-sigs/cri-tools/pull/615#discussion_r431968335).

Sounds good, thank you for the notice.

File: pkg/common/file.go (hunk 3 above)
Comment (hickeyma) [reply pkg/common/file.go:88]: Resolved, incorrect tag on my part.

**Comment by hickeyma** [comment]
Ready for review.

**Comment by hickeyma** [comment]
Thanks for the reviews @saschagrunert and @mikebrow. Thanks for merging also @saschagrunert

**Pull request merged**